### PR TITLE
Update asynchronous_search descriptions

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
### Description
_Describe what this change achieves._

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
Looks like Vale is rightfully unhappy.

**Comment by Naarcha-AWS** [comment]
Pulled the descriptions from docs that have been reviewed by editorial. I'll make Vale happy really fast.

File: spec/namespaces/asynchronous_search.yaml [hunk 1]
```diff
@@ -64,25 +64,25 @@ components:
...
       in: query
-      description: The amount of time that you plan to wait for the results.
+      description: The amount of time that you plan to wait for the results. You can poll the remaining results based on an ID. The maximum value is `300s`. Default is `1s`.
       schema:
         type: string
     asynchronous_search.search::query.keep_on_completion:
       name: keep_on_completion
       in: query
-      description: Whether you want to save the results in the cluster after the search is complete.
+      description: Whether you want to save the results in the cluster after the search is complete. You can examine the stored results at a later time.
```
Comment (Naarcha-AWS) [inline spec/namespaces/asynchronous_search.yaml:73]:
```suggestion
      description: Whether you want to save the results in the cluster after the search is complete. 
```

**Comment by Naarcha-AWS** [comment]
@dblock: This is ready to merge.

**Review by dblock — APPROVED** [review]

**Pull request merged**